packages/compiler/core/logger.ts

```diff
@@ -64,16 +65,42 @@ export function formatLog(log: ProcessedLog): string {
   const location = log.sourceLocation;
   if (location?.file) {
     const formattedLocation = formatSourceLocation(location);
-    return `${formattedLocation} - ${content}`;
+    const sourcePreview = formatSourcePreview(location);
+    return `${formattedLocation} - ${content}${sourcePreview}`;
   } else {
     return content;
   }
 }
 
 export function formatSourceLocation(location: SourceLocation) {
+  const { line, column } = getLineAndColumn(location);
+  const path = location.file.path;
+
+  return `${path}:${line}:${column}`;
+}
+
+/**
+ * Create a preview of where the log location is.
+ *
+ * ----------------------------------------------
+ *   4 |
+ *   5 | @route("/alpha/{id}")
+ * > 6 | op doAlpha(@path id: string): abc;
+ *     |                               ^
+ *   7 |
+ *   8 | @route("/beta/{id}")
+ *   9 | op doBeta(@path id: string): string;
+ */
+function formatSourcePreview(location: SourceLocation) {
+  const { line, column } = getLineAndColumn(location);
+
+  const result = codeFrameColumns(location.file.text, { start: { line, column } });
+  return `\n${result}`;
+}
+
+function getLineAndColumn(location: SourceLocation): { line: number; column: number } {
   const pos = location.file.getLineAndCharacterOfPosition(location.pos ?? 0);
```

Comment:
everytime you use `getLineAndCharacterOfPosition` you have to do +1 to both `line` and `column`. Is there a reason those are 0 based when the lines and column number are 1 based? Should we just change the implementation of this to be 1 based?

Reply:
make sense will keep it this way then. Would feel worst having to do -1 that +1